nixos/modules/services/web-apps/comfyui.nix

```diff
@@ -0,0 +1,136 @@
+{ config, lib, options, pkgs, ...}:
+
+with lib;
+
+let
+  cfg = config.services.comfyui;
+  defaultUser = "comfyui";
+  defaultGroup = defaultUser;
+  mkComfyUIPackage = cfg: cfg.package.override {
+    torch = cfg.torchPackage;
+    modelsPath = "${cfg.dataPath}/models";
+    inputPath = "${cfg.dataPath}/input";
+    outputPath = "${cfg.dataPath}/output";
+    customNodesPath = "${cfg.dataPath}/custom_nodes";
+  };
+in
+{
+  options = {
+    services.comfyui = {
+      enable = mkEnableOption
+        (mdDoc "The most powerful and modular stable diffusion GUI with a graph/nodes interface.");
+
+      dataPath = mkOption {
+        type = types.str;
+        default = "/var/lib/comfyui";
+        description = mdDoc "path to the folders which stores models, custom nodes, input and output files";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.comfyui;
+        defaultText = "pkgs.comfyui";
+        description = mdDoc "ComfyUI base package to use";
+      };
+
+      torchPackage = mkOption {
+        type = types.package;
+        default = pkgs.python311.pkgs.torch;
+        defaultText = "pkgs.python311.pkgs.torch";
+        description = mdDoc "torch package to use";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = defaultUser;
+        example = "yourUser";
+        description = mdDoc ''
+          The user to run ComfyUI as.
+          By default, a user named `${defaultUser}` will be created whose home
+          directory will contain input, output, custom nodes and models.
+        '';
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = defaultGroup;
+        example = "yourGroup";
+        description = mdDoc ''
+          The group to run ComfyUI as.
+          By default, a group named `${defaultUser}` will be created.
+        '';
+      };
+
+      useCPU = mkOption {
+        type = types.bool;
+        default = false; #!config.cudaSupport && !config.rocmSupport;
+        description = mdDoc ''
+          Uses the CPU for everything. Very slow, but needed if there is no hardware acceleration.
+        '';
+      };
+
+      port = mkOption {
+        type = types.int;
+        default = 8188;
+        description = mdDoc "Set the listen port for the Web UI and API.";
+      };
+
+      extraArgs = mkOption {
```

Comment:
This could be `types.attrsOf types.str` and merged with the attrset on line 124.
In this way the option would allow for better composability.